src/internal/plugins/controllers/drawers.ts

```diff
@@ -22,25 +23,33 @@ export interface DrawerConfig {
   };
   mountContent: (container: HTMLElement) => void;
   unmountContent: (container: HTMLElement) => void;
+  keepContentMounted?: boolean;
+  onShow?: NonCancelableEventHandler;
+  onHide?: NonCancelableEventHandler;
 }
 
 export type UpdateDrawerConfig = Pick<DrawerConfig, 'id' | 'badge' | 'resizable' | 'defaultSize'>;
 
 export type DrawersRegistrationListener = (drawers: Array<DrawerConfig>) => void;
 
+export type DrawersOpenedListener = (drawerId: string) => void;
+
 export interface DrawersApiPublic {
   registerDrawer(config: DrawerConfig): void;
   updateDrawer(config: UpdateDrawerConfig): void;
+  openDrawer(drawerId: string): void;
```

Comment:
I think, it is a good time to go ahead of our current requirements and expose `closeDrawer(id)` as well